Mod Configuration Menu/Mods/BG3MCM/ScriptExtender/Lua/Client/Components/FrameManager.lua

```diff
@@ -0,0 +1,135 @@
+ ---@class FrameManager: MetaClass
+FrameManager = _Class:Create("FrameManager", nil, {
+    menuCell = nil,
+    contentCell = nil,
+    contentGroups = {}
+})
+
+
+function FrameManager:initFrameLayout(parent)
+    local layoutTable = parent:AddTable("MenuAndContent", 2)
+    layoutTable:AddColumn("Menu", "WidthFixed")
+    layoutTable:AddColumn("Frame", "WidthStretch")
+    local row = layoutTable:AddRow()
+    self.menuCell = row:AddCell()
+    self.menuCell.ItemWidth = 100
+    self.contentCell = row:AddCell()
+end
+
+---@param guidToShow string
+function FrameManager:setVisibleFrame(guidToShow)
+    for uuid, group  in pairs(self.contentGroups) do
+        group.Visible = (guidToShow == uuid)
+    end
+end
+
+---@param text string
+function FrameManager:AddMenuSection(text)
+    self.menuCell:AddSeparatorText(text)
+end
+
+---@param textButton string
+---@param tooltipText string
+---@param uuid string
+---@return any the group to add Content associated to the button
+function FrameManager:addButtonAndGetModTabBar(textButton, tooltipText, modGUID)
+    if not textButton or not modGUID then
+        MCMWarn(0, "addButtonAndGetModTabBar called with invalid parameters")
+        return
+    end
+
+    textButton = UIStyle:Wrap(textButton, 33)
+
+    local menuButton = self:CreateMenuButton(self.menuCell, textButton, modGUID)
+    if tooltipText then
+        self:AddTooltip(menuButton, tooltipText, modGUID)
+    end
+    local uiGroupMod = self.contentCell:AddGroup(modGUID)
+    self.contentGroups[modGUID] = uiGroupMod
+
+    local modTabs = uiGroupMod:AddTabBar(modGUID .. "_TABS")
+    modTabs.IDContext = modGUID .. "_TABS"
+
+    return modTabs
+end
+
+
+ ---@param uuid string
+ ---@return any the group Content associated to uuid
+ function FrameManager:GetGroup(uuid)
+     return self.contentGroups[uuid]
+ end
+
+ ---@param uuid string
+ ---@return any the group Content associated to uuid
+ function FrameManager:GetModTabBar(uuid)
+     return self.contentGroups[uuid].Children[1]
+ end
```

Comment:
Apart from not handling nils, the problem with accessing `Children[1]` is that it assumes the first child is the mod tab bar we're interested in. However, the position of elements in a table, if the order of children change or if the desired child is not always at index 1, this approach can lead to incorrect or unexpected behavior, as I have already experience by trying to add stuff to `uiGroupMod`.
I think `contentGroup.Children` should then instead be iterated with a check whether it's the mod tab bar.

Reply:
line 50 a, i create tabBar first always

Reply:
Yes, but that can be broken if that ever changes